themes/default/content/registry/packages/aws/how-to-guides/bucketv2-migration.md

```diff
@@ -0,0 +1,2423 @@
+                .build())
+            .build(), CustomResourceOptions.builder()
+                .dependsOn(myBucketOwnership)
+                .build());
+
+    }
+}
+
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+name: example
+runtime: yaml
+variables:
+  currentUser:
+    fn::invoke:
+      function: aws:s3:getCanonicalUserId
+      arguments: {}
+resources:
+  my-bucket:
+    type: aws:s3:BucketV2
+  my-bucket-ownership:
+    type: aws:s3:BucketOwnershipControls
+    properties:
+      bucket: ${my-bucket.id}
+      rule:
+        objectOwnership: "BucketOwnerPreferred"
+  my-bucket-acl:
+    type: aws:s3:BucketAclV2
+    properties:
+      bucket: ${my-bucket.bucket}
+      accessControlPolicy:
+        owner:
+          id: ${currentUser.id}
+        grants:
+          - permission: "READ"
+            grantee:
+              type: "CanonicalUser"
+              id: ${currentUser.id}
+    options:
+      dependsOn:
+        - ${my-bucket-ownership}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+
+
+
+Note that `dependsOn` on `BucketOwnershipControls` is required for Pulumi to correctly order the operations for this
+infrastructure. Consult the documentation on `aws.s3.BucketAclV2` for more fully worked examples of configuring ACL.
+
+### hostedZoneId input
+
+This will only be available as an output. If your program specified this as an input, simply remove it. The input was
+ignored by prior versions of the provider and was exposed in error.
+
+### other inputs
+
+All other removed inputs are treated similarly. Consult the documentation for the matching resource replacing the input
+for more information:
+
+| Input                    | Resource                                   |
+|--------------------------|--------------------------------------------|
+| lifecycleRules           | aws.s3.BucketLifecycleConfigurationV2      |
+| loggings                 | aws.s3.BucketLoggingV2                     |
+| objectLockConfiguration  | aws.s3.BucketObjectLockConfigurationV2     |
+| replicationConfiguration | aws.s3.BucketReplicationConfig             |
+| requestPayer             | aws.s3.BucketRequestPaymentConfigurationV2 |
+| versionings              | aws.s3.BucketVersioningV2                  |
+| website                  | aws.s3.BucketWebsiteConfigurationV2        |
+| websiteDomain            | aws.s3.BucketWebsiteConfigurationV2        |
+| websiteEndpoint          | aws.s3.BucketWebsiteConfigurationV2        |
+
+
+## Avoiding replacement
+
+In situations when replacing the bucket in the AWS account is not acceptable, it is possible to perform a manual
+migration that changes Pulumi program and state to track buckets using the new resource without executing any changes
+against the actual cloud account. While the details will vary depending on your use case, this procedure generally
+involves the following steps:
+
+- Find URNs for legacy Bucket Pulumi resources using `pulumi stack export`
```

Comment:
We should be able to extract those with jq, right? Could be a nice addition to add that here